### PR TITLE
DT-1023: ✨️ Add in transfer to savings method and validation

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/service/FinanceService.java [hunk 1]
```diff
@@ -1,29 +1,110 @@
...
+    private final OffenderTrustAccountRepository offenderTrustAccountRepository;
 
     @VerifyBookingAccess(overrideRoles = {"SYSTEM_USER", "GLOBAL_SEARCH"})
     public Account getBalances(final Long bookingId) {
         return bookingRepository.getBookingAgency(bookingId)
                 .map(agency -> financeRepository.getBalances(bookingId, agency))
                 .orElse(null);
     }
+
+    // @Transactional
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/service/FinanceService.java:46]: method not doing anything at present, so read only will do for now

File: src/main/java/uk/gov/justice/hmpps/prison/repository/FinanceRepository.java [hunk 2]
```diff
@@ -1,8 +1,21 @@
...
 import uk.gov.justice.hmpps.prison.api.model.Account;
 
+import java.math.BigDecimal;
+import java.util.Date;
+
 public interface FinanceRepository {
 
     Account getBalances(long bookingId, final String agencyId);
+
+    void insertIntoOffenderTrans(final String prisonId, final long offId, final long offBookId,
```
Comment (mikehalmamoj) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/FinanceRepository.java:13]: I don't think `final` is part of the method signature and is therefore redundant?

File: src/main/java/uk/gov/justice/hmpps/prison/util/ResourceUtils.java [hunk 3]
```diff
@@ -1,9 +1,18 @@
...
+
 public enum ResourceUtils {
     INSTANCE;
 
     public static <T> T nvl(final T assertNullVal, final T defaultVal) {
         return assertNullVal != null ? assertNullVal : defaultVal;
     }
+
+    public static String getUniqueClientId(final String clientName, final String clientUniqueRef) {
+        if (StringUtils.isBlank(clientUniqueRef)) {
```
Comment (mikehalmamoj) [inline src/main/java/uk/gov/justice/hmpps/prison/util/ResourceUtils.java:13]: Is this required - already tested in the API.  If so, better to throw another `ValidationException`?

File: src/test/java/uk/gov/justice/hmpps/prison/api/model/TransferTransactionTest.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+import org.junit.Test;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.Validation;
+import javax.validation.Validator;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TransferTransactionTest {
+    private static final Validator validator = Validation.buildDefaultValidatorFactory().getValidator();
```
Comment (mikehalmamoj) [inline src/test/java/uk/gov/justice/hmpps/prison/api/model/TransferTransactionTest.java:12]: I haven't seen this before.  Very nice.

File: src/main/java/uk/gov/justice/hmpps/prison/util/ResourceUtils.java (hunk 3 above)
Comment (petergphillips) [reply src/main/java/uk/gov/justice/hmpps/prison/util/ResourceUtils.java:13]: well this is a refactor from `NomisApiV1ResourceImpl`.  In `NomisApiV1Resource.getHolds` it is optional so can be null, which is therefore this scenario

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**